src/python/visualize_matching_result.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import cv2
+import argparse
+from pathlib import Path
+import protos_io
+
+
+def createHiddenMatchImage(width, height):
+    image = np.zeros((height, width, 3), dtype=np.uint8)
+    text_left_corner = (int(width / 3), int(height / 2))
+    image = cv2.putText(
+        image,
+        "No match found",
+        text_left_corner,
+        cv2.FONT_HERSHEY_SIMPLEX,
+        1,
+        (255, 255, 255),
+        2,  # thickness
+    )
+    return image
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--matching_result",
+        required=True,
+        type=Path,
+        help="Path to the matching result .MatchingResult.pb file",
+    )
+    parser.add_argument(
+        "--query_images",
+        required=True,
+        type=Path,
+        help="Path to directory with query images.",
+    )
+    parser.add_argument(
+        "--reference_images",
+        required=True,
+        type=Path,
+        help="Path to directory with reference images.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        required=True,
+        type=Path,
+        help="Output dir for the resulting images.",
+    )
+
+    args = parser.parse_args()
+
+    matching_result = protos_io.read_matching_result(args.matching_result)
+
+    args.output_dir.mkdir(exist_ok=True)
```

Comment:
nit: I think it would be nice to warn the user if their output folder with images is going to be overwritten.

Reply:
Done, thanks.